src/Plugin/ContainerAwarePluginManager.php

```diff
@@ -61,8 +61,7 @@ public function hasDefinition($plugin_id) {
   public function createInstance($plugin_id, array $configuration = array()) {
     // @todo: Use ->expandArguments() when get() disallows getting private
     //        services.
-    $plugin = clone $this->container->get($this->servicePrefix . $plugin_id);
-    return $plugin;
+    return $this->container->createInstance($this->servicePrefix . $plugin_id, $configuration);
```

Comment:
Call this with the $definition instead, get the definition via: $this->getDefinition($plugin_id);